### PR TITLE
Modified PX Backup upgrade TC to include partial success validation

### PR DESCRIPTION
**Pull request by sabrarhussaini**
Modified the TC **PXBackupEndToEndBackupAndRestoreWithUpgrade** to include additional steps for validating the backups with Partial Success.

**Which issue(s) this PR fixes** (optional)
Closes #PB-7198

**Special notes for your reviewer**:
Only the newly added steps could be verified due to the a bug:
Upgrade path: 2.6.0 -> 2.7.2
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5963/
Upgrade path: 2.7.0 -> 2.7.2
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5967/
Upgrade path: 2.7.1 -> 2.7.2
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5956/

### CODE REVIEW

File: tests/backup/backup_upgrade_test.go [hunk 1]
```diff
@@ -519,7 +647,8 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", Label(TestCase
 			err := PxBackupUpgrade(LatestPxBackupVersionFromEnv)
 			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying completion of px-backup upgrade to latest version [%s]", LatestPxBackupVersionFromEnv))
 			// Stork Version will be upgraded on both source and destination if env variable TARGET_STORK_VERSION is defined.
-			targetStorkVersion := os.Getenv("TARGET_STORK_VERSION")
+			// targetStorkVersion := os.Getenv("TARGET_STORK_VERSION")
+			targetStorkVersion := "24.3.0-dev"
```
Comment (mkoppal-px) [inline tests/backup/backup_upgrade_test.go:651]: Remove this hard-coding before merge

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Comment by mkoppal-px** [comment]
Give a full run without commenting out any steps

File: tests/backup/backup_upgrade_test.go [hunk 2]
```diff
@@ -368,6 +401,29 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", Label(TestCase
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup location [%s] with UID [%s] using the bucket [%s]", backupLocationName, backupLocationUid, bucketName))
 			}
 		})
+
+		Step("Create cloud credentials and backup locations for partial backup validation", func() {
```
Comment (ak-px) [inline tests/backup/backup_upgrade_test.go:405]: Why we need separate backup location here?

File: tests/backup/backup_upgrade_test.go [hunk 3]
```diff
@@ -595,6 +724,38 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", Label(TestCase
...
+			log.InfoD("Restoring the scheduled backup with partial success")
+			namespaceMapping := make(map[string]string)
+			for _, namespace := range partialAppNamespaces {
+				namespaceMapping[namespace] = namespace + RandomString(4)
+			}
+			k8sCore := k8score.Instance()
+			log.FailOnError(err, "Getting the failed PVC list")
+			pvcList, err := k8sCore.GetPersistentVolumeClaims(partialAppNamespaces[0], nil)
+			for _, pvc := range pvcList.Items {
+				failedVolumes = append(failedVolumes, pvc.Name)
```
Comment (ak-px) [inline tests/backup/backup_upgrade_test.go:746]: May need to rebase the latest changes for failedVolumes

Comment (sabrarhussaini) [reply tests/backup/backup_upgrade_test.go]: Done

File: tests/backup/backup_upgrade_test.go (hunk 2 above)
Comment (sabrarhussaini) [reply tests/backup/backup_upgrade_test.go]: The plan was to not to touch or make use of any of the existing backup objects.

File: tests/backup/backup_upgrade_test.go (hunk 1 above)
Comment (sabrarhussaini) [reply tests/backup/backup_upgrade_test.go]: Done.

File: tests/backup/backup_upgrade_test.go [hunk 4]
```diff
@@ -820,10 +993,15 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", Label(TestCase
 
 	})
 	JustAfterEach(func() {
-		allContexts := append(srcClusterContexts, destClusterContexts...)
-		defer EndPxBackupTorpedoTest(allContexts)
+		defer EndPxBackupTorpedoTest(partialAppContexts)
```
Comment (ak-px) [inline tests/backup/backup_upgrade_test.go:996]: who about having single instance of EndPxBackupTorpedoTest with allContexts

Comment (sabrarhussaini) [reply tests/backup/backup_upgrade_test.go]: Done, please review the changes.

**Review by ak-px — APPROVED** [review]
LGTM

**Pull request merged**